mypy/plugins/common.py

```diff
@@ -113,6 +113,27 @@ def add_method_to_class(
 ) -> None:
     """Adds a new method to a class definition.
     """
+    function_type = api.named_type('__builtins__.function')
+    add_method_to_class_with_function_type(function_type,
```

Comment:
Format this differently, please. I'd suggested putting `function_type` on the second line, and indenting everything one indentation.

Reply:
Oh, but actually, I would just skip creating a new function and add `function_type` as an optional argument to `add_method_to_class`

Reply:
> Oh, but actually, I would just skip creating a new function and add `function_type` as an optional argument to `add_method_to_class`

I thought about that at first, but the problem is that we'd need to pass in an argument for the semantic analyzer interface, which we wouldn't have access to from the type checking hooks in singledispatch.py. I guess we could just pass in `None` for `api` and add a `type: ignore` comment but this seemed cleaner.

> Format this differently, please. I'd suggested putting `function_type` on the second line, and indenting everything one indentation.

Ok, I'll fix this.

Reply:
The formatting should be fixed in 29f8a8c1dd4bcdf3f79ed85e9393e4986818ba9d. Let me know if you meant something else by making `function_type` an optional argument to `add_method_to_class`.